src/data/fin/interval.lean

```diff
@@ -20,6 +20,14 @@ variables (n : ℕ)
 
 instance : locally_finite_order (fin n) := subtype.locally_finite_order _
 
+instance : Π n, locally_finite_order_top (fin n)
+| 0 := infer_instance
+| (n + 1) := infer_instance
+
+instance : Π n, locally_finite_order_bot (fin n)
+| 0 := infer_instance
+| (n + 1) := infer_instance
```

Comment:
Can you add a test to verify that this doesn't create diamonds?

Reply:
Do we really need a test for this? I just checked and it's indeed refl.